packages/theme/src/cli/utilities/theme-environment/theme-environment.ts

```diff
@@ -72,9 +70,18 @@ interface DevelopmentServerInstance {
   close: () => Promise<void>
 }
 
-function createDevelopmentServer(theme: Theme, ctx: DevServerContext) {
+function createDevelopmentServer(theme: Theme, ctx: DevServerContext, initialWork?: Promise<void>) {
   const app = createApp()
 
+  if (initialWork) {
+    app.use(
+      defineLazyEventHandler(async () => {
+        await initialWork
+        return defineEventHandler(() => {})
+      }),
+    )
+  }
+
```

Comment:
This has nothing to do with the original PR, just a small perf improvement I forgot in another PR 😅